src/domain/fx/cyril.js

```diff
@@ -0,0 +1,466 @@
+      const proxyParticipantAccountDetails = await ProxyCache.getProxyParticipantAccountDetails(transfer.payeeFsp, sendingFxpRecord.targetCurrency)
+      if (proxyParticipantAccountDetails.participantCurrencyId) {
+        let isPositionChange = false
+        if (proxyParticipantAccountDetails.inScheme) {
+          isPositionChange = true
+        } else {
+          // We are not expecting this. Payee participant is a proxy and have an account in the targetCurrency.
+          // In this case we need to check if FXP is also a proxy and have the same account as payee.
+          const proxyParticipantAccountDetails2 = await ProxyCache.getProxyParticipantAccountDetails(sendingFxpRecord.counterPartyFspName, sendingFxpRecord.targetCurrency)
+          if (!proxyParticipantAccountDetails2.inScheme && (proxyParticipantAccountDetails.participantCurrencyId !== proxyParticipantAccountDetails2.participantCurrencyId)) {
+            isPositionChange = true
+          }
+        }
+        if (isPositionChange) {
+          result.positionChanges.push({
+            isFxTransferStateChange: false,
+            transferId,
+            participantCurrencyId: proxyParticipantAccountDetails.participantCurrencyId,
+            amount: -sendingFxpRecord.targetAmount
+          })
+        }
+      }
+    } else if (receivingFxpExists) {
+      // If we have a receiving FXP, Create obligation between debtor party to the transfer and FXP in currency of transfer
+      const proxyParticipantAccountDetails = await ProxyCache.getProxyParticipantAccountDetails(receivingFxpRecord.counterPartyFspName, receivingFxpRecord.sourceCurrency)
+      if (proxyParticipantAccountDetails.participantCurrencyId) {
+        let isPositionChange = false
+        if (proxyParticipantAccountDetails.inScheme) {
+          isPositionChange = true
+        } else {
+          // We are not expecting this. FXP participant is a proxy and have an account in the sourceCurrency.
+          // In this case we need to check if Payer is also a proxy and have the same account as FXP.
+          const proxyParticipantAccountDetails2 = await ProxyCache.getProxyParticipantAccountDetails(transfer.payerFsp, receivingFxpRecord.sourceCurrency)
+          if (!proxyParticipantAccountDetails2.inScheme && (proxyParticipantAccountDetails.participantCurrencyId !== proxyParticipantAccountDetails2.participantCurrencyId)) {
+            isPositionChange = true
+          }
+        }
+        if (isPositionChange) {
+          result.positionChanges.push({
+            isFxTransferStateChange: true,
+            commitRequestId: receivingFxpRecord.commitRequestId,
+            participantCurrencyId: proxyParticipantAccountDetails.participantCurrencyId,
+            amount: -receivingFxpRecord.sourceAmount
+          })
+        }
+      }
+    }
+
+    // TODO: Remove entries from watchlist
```

Comment:
still pending?

Reply:
@vijayg10 Can this be removed/moved to a story. Not sure of context.

Reply:
Removed.